rules/sort-interfaces.ts

```diff
@@ -81,252 +26,8 @@ let defaultOptions: Required<Options<string[]>[0]> = {
   groups: [],
 }
 
-export default createEslintRule<Options<string[]>, MESSAGE_ID>({
-  create: context => ({
-    TSInterfaceDeclaration: node => {
```

Comment:
I'm not sure we should get rid of code duplication here. Could this cause us any problems in the future?

Reply:
@azat-io

> Could this cause us any problems in the future?

There shouldn't be:
- Interface nodes (`TSInterfaceDeclaration`) have a `body.body` field which is typed as `TypeElement[]`.
- Object type nodes (`TSTypeLiteral`) have `members` field which is typed as `TypeElement[]`.

Each object type/interface member acts the same way on a ESLint parsing level.

If we decide to add a feature solely for one rule and not the other (I don't have a use case for this in mind), then we can also just add a parameter to the common `sortObjectTypeElements` function.

I can still copy-paste the code in each rule if you feel like it's better, it's just that we need not to forget to update it in two places when adding a feature to one of the rules.